pkg/eventbus/event_bus.go

```diff
@@ -35,19 +47,42 @@ func (eb *eventbus) Start(c *context.Context) {
 	// no need to call TopicInit now, we have fixed topic
 	eb.context = c
 
-	mqttURL := config.CONFIG.GetConfigurationByKey("mqtt.server")
 	nodeID := config.CONFIG.GetConfigurationByKey("edgehub.controller.node-id")
-	if mqttURL == nil || nodeID == nil {
+	if nodeID == nil {
 		panic("mqtt url or node id not configured")
 	}
-	hub := &mqttBus.MQTTClient{
-		MQTTUrl: mqttURL.(string),
-	}
-	mqttBus.MQTTHub = hub
+
 	mqttBus.NodeID = nodeID.(string)
 	mqttBus.ModuleContext = c
-	hub.InitSubClient()
-	hub.InitPubClient()
+
+	if eb.mqttMode >= doubleMqttMode {
+		// launch an external mqtt server
+		externalMqttUrl := config.CONFIG.GetConfigurationByKey("mqtt.external-mqtt")
+		if externalMqttUrl == nil {
+			panic("third party mqtt url not configured")
+		}
+
+		hub := &mqttBus.MQTTClient{
+			MQTTUrl: externalMqttUrl.(string),
+		}
+		mqttBus.MQTTHub = hub
+		hub.InitSubClient()
+		hub.InitPubClient()
+	}
+
+	if eb.mqttMode <= doubleMqttMode {
+		internalMqttURL := config.CONFIG.GetConfigurationByKey("mqtt.internal-mqtt")
+		if internalMqttURL == nil {
+			panic("mqtt url is not configured")
```

Comment:
pannic info : "internal mqtt url is not configured "will be more helpful when to identify problem.